examples/app-basic/src/components/SecretInput/SecretInput.tsx

```diff
@@ -0,0 +1,25 @@
+import React, { HTMLProps } from 'react';
+import { Input, HorizontalGroup, Button } from '@grafana/ui';
+
+type Props = Omit<HTMLProps<HTMLInputElement>, 'prefix' | 'size' | 'width' | 'ref'> & {
+  /** Defines if the password is already set or not. (It is needed as in some cases the backend doesn't send back the actual password) */
+  isConfigured: boolean;
+  /** Called when the user clicks on the "Reset" button */
+  onReset: () => void;
+};
+
+// This replaces the "LegacyForms.SecretFormField" component from @grafana/ui, so we can start using the newer form components.
+export const SecretInput = ({ isConfigured, onReset, ...props }: Props) => {
```

Comment:
We should probably open a PR to add this to grafana instead of having a copy like this.

Reply:
but I think we can leave it like this for now. We have an issue to do it here: https://github.com/grafana/grafana/issues/45163